src/LazyValue.php

```diff
@@ -19,27 +19,47 @@ final class LazyValue
     /** @var callable():mixed */
     private $factory;
 
+    /** @var bool */
+    private $memoize;
+
+    /** @var mixed */
+    private $memoizedValue = null;
+
     /**
      * @param callable():mixed $factory
      */
-    public function __construct(callable $factory)
+    public function __construct(callable $factory, bool $memoize = false)
     {
         $this->factory = $factory;
+        $this->memoize = $memoize;
+    }
+
+    public static function memoize(callable $factory): self
```

Comment:
Should we add a named constructor `new` for non-memoized? Would make it consistent, but we can't make the contstructor private for BC.

Reply:
I think this is a good idea, maybe we could add  a depercation layer for using `new Lazy()` directly?
see https://github.com/zenstruck/foundry/blob/1.x/src/FactoryCollection.php#L41-L43 as an example